src/lib/routes/admin-api/project/project-features.ts

```diff
@@ -907,28 +910,16 @@ export default class ProjectFeaturesController extends Controller {
         const strategy = await this.featureService.getStrategy(strategyId);
 
         const { newDocument } = applyPatch(strategy, patch);
-        const updatedStrategy = await this.featureService.updateStrategy(
-            strategyId,
-            newDocument,
-            { environment, projectId, featureName },
-            userName,
-            req.user,
+        const updatedStrategy = await this.startTransaction(async (tx) =>
+            this.transactionalFeatureToggleService(tx).updateStrategy(
+                strategyId,
+                newDocument,
+                { environment, projectId, featureName },
+                userName,
+                req.user,
+            ),
         );
-        const feature = await this.featureService.getFeature({ featureName });
 
-        const env = feature.environments.find((e) => e.name === environment);
-        const hasOnlyDisabledStrategies = env!.strategies.every(
```

Comment:
the ! to ignore type system in production code is not a good idea. I replaced it with env?. because I managed to get null pointer here